service/worker/sysworkflow/initiator.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package sysworkflow
+
+import (
+	"context"
+	"github.com/uber/cadence/client/frontend"
+	"go.uber.org/cadence/client"
+	"time"
+)
+
+type (
+
+	// Initiator is used to trigger system tasks
+	Initiator interface {
+		Archive(request *ArchiveRequest) error
+	}
+
+	initiator struct {
+		cadenceClient  client.Client
+		sysWorkflowIDs []string
+	}
+
+	// Signal is the data sent to system tasks
+	Signal struct {
+		RequestType    RequestType
+		ArchiveRequest *ArchiveRequest
+	}
+
+	// ArchiveRequest signal used for archival task
+	ArchiveRequest struct {
+		UserWorkflowID string
+		UserRunID      string
+	}
+)
+
+// NewInitiator creates a new Initiator
+func NewInitiator(frontendClient frontend.Client, numSysWorkflows int) Initiator {
+	return &initiator{
+		cadenceClient:  client.NewClient(frontendClient, Domain, &client.Options{}),
+		sysWorkflowIDs: RandomSlice(numSysWorkflows),
+	}
+}
+
+// Archive starts an archival task
+func (i *initiator) Archive(request *ArchiveRequest) error {
+	workflowID := PickRandom(i.sysWorkflowIDs)
```

Comment:
I'm a little confused by this pattern.  Won't this create new workflow IDs for system workflow each time you restart the server?  How do you control the number of concurrent system workflow execution?
It would be good to put a comment around expectation here.